emukit/bayesian_optimization/acquisitions/max_value_entropy_search.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+from typing import Union, Callable
+
+import scipy
+import numpy as np
+from scipy.stats import norm
+from scipy.optimize import bisect
+
+from ...core.acquisition import Acquisition
+from ...core.interfaces import IModel
+from ...core.parameter_space import ParameterSpace
+
+from ..interfaces import IEntropySearchModel
+from ...experimental_design import RandomDesign
+
+
+
+class MaxValueEntropySearch(Acquisition):
+
+    def __init__(self, model: Union[IModel, IEntropySearchModel], space: ParameterSpace,
+                 num_samples: int = 10, grid_size: int = 5000) -> None:
+
+        """
+        MES acquisition function approximates the distribution of the value at the global
+        minimum and tries to decrease its entropy. See this paper for more details:
+        Z. Wang, S. Jegelka
+        Max-value Entropy Search for Efficient Bayesian Optimization
+        ICML 2017
+
+        This part of the code is heavily based on 
+        https://github.com/GPflow/GPflowOpt/blob/master/gpflowopt/acquisition/mes.py
+
+        :param model: GP model to compute the distribution of the minimum dubbed pmin.
+        :param space: Domain space which we need for the sampling of the representer points
+        :param num_samples: integer determining how many samples to draw of the minimum (does not need to be large)
+        :grid_size: number of random locations in grid used to fit the gumbel distribution and approximately generate the samples of the minimum (recommend scaling with problem dimension, i.e. 10000*d)
+        """
+        super().__init__()
+
+        if not isinstance(model, IEntropySearchModel):
+            raise RuntimeError("Model is not supported for MES")
+
+        self.model = model
+        self.space = space
+        self.num_samples = num_samples
+        self.grid_size = grid_size
+
+        # Initialize parameters to lazily compute them once needed
+        self.mins = None
+    def update_parameters(self):
+        self._prepare_sample()
```

Comment:
I am not sure there is any point in having this private method. I would just put the whole thing into "update_parameters" and use it as needed. Is there any reason you did it this way?

Reply:
I was copying the format of your entropy search acquisition . I can combine if you want?

Reply:
Frankly i missed that in the original one. I believe combining is a good idea, if you agree, please go ahead with that

Reply:
Done